framework/include/transfers/MultiAppTransfer.h

```diff
@@ -40,13 +40,29 @@ class MultiAppTransfer : public Transfer
   void variableIntegrityCheck(const AuxVariableName & var_name) const;
 
   /// Return the MultiApp that this transfer belongs to
-  const std::shared_ptr<MultiApp> getMultiApp() const { return _multi_app; }
+  const std::shared_ptr<MultiApp> getMultiApp() const
+  {
+    if (_from_multi_app && _to_multi_app)
+      mooseError("Unclear which app you want from Transfer ", name());
+    else if (_from_multi_app)
+      return _from_multi_app;
+    else if (_to_multi_app)
+      return _to_multi_app;
+    else if (_multi_app)
+      return _multi_app;
+    else
+      mooseError("Should not get here, there should be a multiapp");
+  }
 
   /// Return the execution flags, handling "same_as_multiapp"
   virtual const std::vector<ExecFlagType> & execFlags() const;
 
 protected:
-  /// The MultiApp this Transfer is transferring data to or from
+  /// The MultiApps this Transfer is transferring data to or from
+  std::shared_ptr<MultiApp> _from_multi_app;
+  std::shared_ptr<MultiApp> _to_multi_app;
```

Comment:
Should we make these private?

Reply:
also it would break all the apps unless we get a round of patches in before, and we still need to get a round of patches in after to fix the input files

Reply:
it's getters for the sake of getters at this point

Reply:
What I meant was that move `_from_multi_app` and `_to_multi_app` to `private` , and then add getter.

I just do not want users to use raw members that make integration harder in the future

Reply:
Yeah I know what you mean. This is inherited from the `_multi_app` attribute not being private

Reply:
_multi_app is not private because we made a mistake. We are trying to avoid making the same mistake again.

But, anyway, the conversation diverged.